kernprof.py

```diff
@@ -155,53 +155,46 @@ def find_script(script_name):
-            "'@profile' to decorate a single function, or 'with profile:' "
-            "to profile a single section of code.")
-    parser.add_option('-o', '--outfile', default=None,
-        help="Save stats to <outfile>")
-    parser.add_option('-s', '--setup', default=None,
+def main():
+    def positive_float(value):
+        val = float(value)
+        if val <= 0:
+            raise ArgumentError
+        return val
+
+    parser = ArgumentParser()
+    parser.add_argument('-V', '--version', action='version', version=__version__)
+    parser.add_argument('-l', '--line-by-line', action='store_true',
+        help="Use the line-by-line profiler instead of cProfile. Implies --builtin.")
+    parser.add_argument('-b', '--builtin', action='store_true',
+        help="Put 'profile' in the builtins. Use 'profile.enable()'/'.disable()', "
+            "'@profile' to decorate functions, or 'with profile:' to profile a "
+            "section of code.")
+    parser.add_argument('-o', '--outfile',
+        help="Save stats to <outfile> (default: 'scriptname.lprof' with "
+            "--line-by-line, 'scriptname.prof' without)")
+    parser.add_argument('-s', '--setup',
         help="Code to execute before the code to profile")
-    parser.add_option('-v', '--view', action='store_true',
-        help="View the results of the profile in addition to saving it.")
-    parser.add_option('-u', '--unit', default='1e-6', type=float,
-        action='callback', callback=strictly_positive,
-        help="Output unit (in seconds) in which the timing info is to be "
-            "displayed (for --view). Defaults to 1e-6.")
-    parser.add_option('--skip-zero', action='store_true',
-        help="Hide functions which have not been called (for --view).")
+    parser.add_argument('-v', '--view', action='store_true',
+        help="View the results of the profile in addition to saving it")
+    parser.add_argument('-u', '--unit', default='1e-6', type=positive_float,
+        
+        help="Output unit (in seconds) in which the timing info is "
+        "displayed (default: 1e-6)")
+    parser.add_argument('-z', '--skip-zero', action='store_true',
+        help="Hide functions which have not been called")
 
-    if not sys.argv[1:]:
-        parser.print_usage()
-        sys.exit(2)
+    parser.add_argument('script', help="The python script file to run")
+    parser.add_argument('args', nargs='...', help="Optional script arguments")
 
-    options, args = parser.parse_args()
+    options = parser.parse_args()
 
     if not options.outfile:
-        if options.line_by_line:
-            extension = 'lprof'
-        else:
-            extension = 'prof'
-        options.outfile = '%s.%s' % (os.path.basename(args[0]), extension)
+        extension = 'lprof' if options.line_by_line else 'prof'
+        options.outfile = '%s.%s' % (os.path.basename(options.script), extension)
 
 
-    sys.argv[:] = args
+    sys.argv = [options.script, *options.args]
```

Comment:
Are we dropping Python 2.7 support in this next release? That would have to be done to introduce this change.

Reply:
As you want, I can see how to change it so that we keep py27 compatiblity (but we'll have to drop it at some point).

Reply:
I'm willing to drop 2.7 in this release. We just need to update the associated files like `setup.py`, `.travis.yml`, etc... We may as well drop 3.5 if we're going to do that. 

Alternatively, we can keep 2.7 / 3.5 around and write `sys.argv = [options.script] +  options.args`. Its up to you.

Reply:
Fixed